resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -134,6 +134,13 @@ private[spark] class KubernetesClusterSchedulerBackend(
       }
     }
 
+    Utils.tryLogNonFatalError {
+      kubernetesClient
+        .persistentVolumeClaims()
+        .withLabel(SPARK_APP_ID_LABEL, applicationId())
+        .delete()
+    }
```

Comment:
So we don't delete PVC before?

Reply:
Previously, the lifecycle is tied with the executor pod.
Now, the lifecycle is tied with the driver pod. So, it will be deleted when the driver pod die.
This code is to support early deletion at the app termination.
This is the same one for `spark.kubernetes.driver.service.deleteOnTermination`~

Reply:
@dongjoon-hyun shouldn't there be separate property  spark.kubernetes.driver.pvc.deleteOnTermination based on which this been called.